src/backoffice/packages/repository/package.repository.ts

```diff
@@ -0,0 +1,96 @@
+import { UmbPackageStore, UMB_PACKAGE_STORE_TOKEN } from './package.store';
+import { UmbPackageServerDataSource } from './sources/package.server.data';
+import { UmbControllerHostInterface } from '@umbraco-cms/controller';
+import { UmbContextConsumerController } from '@umbraco-cms/context-api';
+import { ManifestBase } from '@umbraco-cms/extensions-registry';
+
+/**
+ * A repository for Packages which mimicks a tree store.
+ * @export
+ */
+export class UmbPackageRepository {
+	#init!: Promise<void>;
+	#packageStore?: UmbPackageStore;
+	#packageSource: UmbPackageServerDataSource;
+
+	constructor(host: UmbControllerHostInterface) {
+		this.#packageSource = new UmbPackageServerDataSource(host);
+		this.#init = new Promise((res) => {
+			new UmbContextConsumerController(host, UMB_PACKAGE_STORE_TOKEN, (instance) => {
+				this.#packageStore = instance;
+				this.#requestRootItems(instance);
+				this.#requestPackageMigrations(instance);
+				res();
+			});
+		});
+	}
+
+	/**
+	 * Request the root items from the Data Source
+	 * @memberOf UmbPackageRepository
+	 * @private
+	 */
+	async #requestRootItems(store: UmbPackageStore) {
```

Comment:
I think we should make these methods public, at some point, so we can request packages on demand without having to make a new repo instance.